typedoc.json

```diff
@@ -0,0 +1,6 @@
+{
+  "entryPoints": ["./src/index.ts"],
+  "excludePrivate": true,
+  "hideGenerator": true,
+  "out": "typedocs"
```

Comment:
I imagine we could extract this configuration to a shared place in the future, so where should generated documentation go? I opted to use `typedocs/` instead of `docs/` in case we ever have a project in the future where we want to manually provide documentation (setup instructions, explanations for concepts, etc.). Open to suggestions.

Reply:
TypeDoc changes removed in [a2ffca7](https://github.com/MetaMask/eth-json-rpc-infura/pull/58/commits/a2ffca7b55ae7bb58a76adce4cc87a0f2bb98488).